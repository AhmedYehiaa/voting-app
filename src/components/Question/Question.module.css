.container {
  position: relative;
  display: inline-block;
  overflow: hidden;
  max-width: 100%;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.16), 0 0 0 1px rgba(0, 0, 0, 0.08);
  height: auto;
}
.container:hover {
  cursor: pointer;
}

.questionCard {
  padding: 15px 15px;
}

.questionCard:hover {
  cursor: pointer;
}

.questionCardTitle {
  text-align: center;
  border-bottom: 2px solid #002837;
  padding-bottom: 10px;
}
.questionCardBody div {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.questionCardBody span {
  font-weight: bold;
}

.voteNow {
  opacity: 0;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.75);
  color: #00cea5;
  font-weight: bold;
  padding: 15px;
  transition: all 0.4s ease-in-out 0s;
}

.voteNow .voteNowText {
  text-align: center;
  font-size: 18px;
  display: inline-block;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.container:hover .voteNow {
  opacity: 1;
}
