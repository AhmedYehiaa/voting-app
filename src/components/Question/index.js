import React from "react";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCalendarDay, faListOl } from "@fortawesome/free-solid-svg-icons";

import styles from "./Question.module.css";

const Question = ({ title, url, publishedAt, choicesLength, history }) => {
  const formateDate = () => {
    return new Date(publishedAt)
      .toDateString()
      .split(" ")
      .slice(1)
      .join(", ");
  };
  return (
    <div className={styles.container} onClick={() => history.push(url)}>
      <div className={styles.questionCard}>
        <h3 className={styles.questionCardTitle} data-testid="title">
          {title}
        </h3>
        <div className={styles.questionCardBody}>
          <div>
            <FontAwesomeIcon icon={faCalendarDay} />
            <p data-testid="publishedAt">{formateDate()}</p>
          </div>
          <div>
            <FontAwesomeIcon icon={faListOl} />
            <p>
              <span data-testid="choicesLength">{choicesLength}</span>
              {"  "}choices
            </p>
          </div>
        </div>
      </div>
      <div className={styles.voteNow}>
        <div className={styles.voteNowText}>VOTE NOW</div>
      </div>
    </div>
  );
};

Question.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  publishedAt: PropTypes.string.isRequired,
  choicesLength: PropTypes.number.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired
};

export default withRouter(Question);
