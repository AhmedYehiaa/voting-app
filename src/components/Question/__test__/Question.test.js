import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter as Router } from "react-router-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

import Question from "../index";

describe("<Question />", () => {
  const question = {
    title: "question title",
    url: "questionUrl/url",
    publishedAt: "2015-05-27T21:22:26.648000+00:00",
    choicesLength: 5
  };

  afterEach(cleanup);

  it("render without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Router>
        <Question
          title={question.title}
          url={question.url}
          publishedAt={question.publishedAt}
          choicesLength={question.choicesLength}
        />
      </Router>,
      div
    );
  });

  it("render question component with the passed props", () => {
    const { getByTestId } = render(
      <Router>
        <Question
          title={question.title}
          url={question.url}
          publishedAt={question.publishedAt}
          choicesLength={question.choicesLength}
        />
      </Router>
    );
    expect(getByTestId("title").textContent).toBe(question.title);
    expect(getByTestId("choicesLength").textContent).toBe(
      String(question.choicesLength)
    );
    expect(getByTestId("publishedAt").textContent).toBe("May, 27, 2015");
  });

  it("matches snapshot", () => {
    const tree = renderer
      .create(
        <Router>
          <Question
            title={question.title}
            url={question.url}
            publishedAt={question.publishedAt}
            choicesLength={question.choicesLength}
          />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
