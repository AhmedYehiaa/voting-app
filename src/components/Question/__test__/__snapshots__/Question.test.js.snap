// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Question /> matches snapshot 1`] = `
<div
  className="container"
  onClick={[Function]}
>
  <div
    className="questionCard"
  >
    <h3
      className="questionCardTitle"
      data-testid="title"
    >
      question title
    </h3>
    <div
      className="questionCardBody"
    >
      <div>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-calendar-day fa-w-14 "
          data-icon="calendar-day"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm64-192c0-8.8 7.2-16 16-16h96c8.8 0 16 7.2 16 16v96c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16v-96zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <p
          data-testid="publishedAt"
        >
          May, 27, 2015
        </p>
      </div>
      <div>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-list-ol fa-w-16 "
          data-icon="list-ol"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M61.77 401l17.5-20.15a19.92 19.92 0 0 0 5.07-14.19v-3.31C84.34 356 80.5 352 73 352H16a8 8 0 0 0-8 8v16a8 8 0 0 0 8 8h22.83a157.41 157.41 0 0 0-11 12.31l-5.61 7c-4 5.07-5.25 10.13-2.8 14.88l1.05 1.93c3 5.76 6.29 7.88 12.25 7.88h4.73c10.33 0 15.94 2.44 15.94 9.09 0 4.72-4.2 8.22-14.36 8.22a41.54 41.54 0 0 1-15.47-3.12c-6.49-3.88-11.74-3.5-15.6 3.12l-5.59 9.31c-3.72 6.13-3.19 11.72 2.63 15.94 7.71 4.69 20.38 9.44 37 9.44 34.16 0 48.5-22.75 48.5-44.12-.03-14.38-9.12-29.76-28.73-34.88zM496 224H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zM16 160h64a8 8 0 0 0 8-8v-16a8 8 0 0 0-8-8H64V40a8 8 0 0 0-8-8H32a8 8 0 0 0-7.14 4.42l-8 16A8 8 0 0 0 24 64h8v64H16a8 8 0 0 0-8 8v16a8 8 0 0 0 8 8zm-3.91 160H80a8 8 0 0 0 8-8v-16a8 8 0 0 0-8-8H41.32c3.29-10.29 48.34-18.68 48.34-56.44 0-29.06-25-39.56-44.47-39.56-21.36 0-33.8 10-40.46 18.75-4.37 5.59-3 10.84 2.8 15.37l8.58 6.88c5.61 4.56 11 2.47 16.12-2.44a13.44 13.44 0 0 1 9.46-3.84c3.33 0 9.28 1.56 9.28 8.75C51 248.19 0 257.31 0 304.59v4C0 316 5.08 320 12.09 320z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <p>
          <span
            data-testid="choicesLength"
          >
            5
          </span>
            
          choices
        </p>
      </div>
    </div>
  </div>
  <div
    className="voteNow"
  >
    <div
      className="voteNowText"
    >
      VOTE NOW
    </div>
  </div>
</div>
`;
